classes/Tools.php

```diff
@@ -1771,7 +1771,7 @@ public static function math_round($value, $places, $mode = PS_ROUND_HALF_UP)
     {
         //If PHP_ROUND_HALF_UP exist (PHP 5.3) use it and pass correct mode value (PrestaShop define - 1)
         if (defined('PHP_ROUND_HALF_UP')) {
-            return round($value, $places, $mode - 1);
+            return round($value ?? 0.0, $places, $mode - 1);
```

Comment:
There will probably be the same issue below when `PHP_ROUND_HALF_UP` is not defined no?
Maybe you should start the function ith `$value = $value ?? 0.0;` so it can fix the whole function in the future (and maybe add a comment to explain the value needs to be "casted" in case invalid null value is used)

Reply:
That said it fixes the last part of the code, but the million dollar question is why is null passed as an argument and is it worth investigating the calling code 🤷‍♂️

Reply:
I think it happens only with behat tests, but I'm not 100% sure how, it's shenanigans related to `getpricestatic`, `priceCalculation` with cached price values...

Reply:
(done !)

Reply:
@matthieu-rolland It should be investigated why there is a null coming into the method.

And regarding the condition, it would be better to catch it at the "entrance" in `ps_round` method. There are other methods called depending on round mode, that also don't accept null.

Reply:
I'll make an issue about this but I won't block the release as long as UI tests and CI is green

Reply:
Well, didn't @matks say that we should not rush anything? I don't like that we put a random hackjob into a method just to supress an error… The null is coming from somewhere and we should find out why.

Could you at least out it to ps_round method so all rounding types are covered and squash the commits? :-)

Reply:
you don't l ike the "hackjob" but you also want to put it everywhere ?

I'm making an issue for this and the behat tests will be fixed there. (I can assign you on this if you're interested)

Reply:
I dont want you to put it everywhere. I am saying that if you look at the code, math_round is only called from ps_round. And depending on a rounding mode, it chooses the rounding method.

So if you change it to different rounding, it will fail on floorf or ceilf for example. Thats why I suggest to at least put the check to ps_round.

Yeah, putting a random check somewhere just to fix tests doesnt seem like a long term solution for me. You will create an issue and the issue will just exist for years.